Notification for Python default rulesets

picimako opened the pull request:
### Changes
- Renamed the `services` package to `rosie` since it contains not only services but rather Rosie related functionality.
- `RosieStartupActivity`
  - Renamed it from `RosieRulesCacheUpdater`.
  - Added the notification popup behaviour: if there is no codiga.yml in the project, the user hasn't clicked "never remind me", and there is at least one Python module in the project, then we show the popup.
- Introduced `CodigaConfigState` as a project service to persist the config to show the popup or not for the given project.

### Notes
- I couldn't find a proper solution that I could use for integration testing, that is why this PR doesn't contain any such test.
- I attached a screenshot of what the popup looks like now:
  - ![notification_popup](https://user-images.githubusercontent.com/5739987/199442188-f3c9eaaa-7d7c-419d-afdd-7aa46f3e74a2.png)
  - The actions are hidden at the moment in a menu, due to they have long texts.

juli1 reviewed (changes requested):
1. if you move `Rosie` from `services` and put it in the package `rosie`, please give an explicit name. We are ending up with `rosie.Rosie` and `rosie.RosieImpl`. Use a name that is more descriptive.
2. I tried to use the feature after pulling the branch and I could not get the popup. Let's make sure it works and if this is a project configuration, we may try another approach.

picimako commented:
- Renamed the mentioned classes.
- Added a subtitle to the notification popup.
- The new logic first checks the project level SDK, then if that is not found or not a Python one, checks the module SDKs. If either of them is a Python SDK, we show the notification.
